Updates related to the animation-composition property (#19940)
mdn/content: files/en-us/web/css/css_animations/using_css_animations/index.md

3. Letting the browser control the animation sequence lets the browser optimize performance and efficiency by, for example, reducing the update frequency of animations running in tabs that aren't currently visible.

## Configuring an animation

To create a CSS animation sequence, you style the element you want to animate with the {{cssxref("animation")}} property or its sub-properties. This lets you configure the timing, duration, and other details of how the animation sequence should progress. This does **not** configure the actual appearance of the animation, which is done using the {{cssxref("@keyframes")}} at-rule as described in the [Defining the animation sequence using keyframes](#defining_the_animation_sequence_using_keyframes) section below.

The sub-properties of the {{cssxref("animation")}} property are:

<!--
- {{cssxref("animation-composition")}}
  - : Specifies the {{Glossary("Composite operation")}} to use when multiple animations affect the same property simultaneously.
-->
- {{cssxref("animation-delay")}}

>   - : Specifies whether an animation should start immediately or add a delay before the start of an animation sequence.

Reviewer: this property also enables starting an animation halfway through. should we include that here?

Author: Got it! Updated the description.

You can optionally include additional keyframes that describe intermediate steps between the start and end of the animation.

## Using the animation shorthand

The {{cssxref("animation")}} shorthand is useful for saving space. As an example, some of the rules we've been using through this article:

```css
p {
  animation-duration: 3s;
  animation-name: slidein;
  animation-iteration-count: infinite;
  animation-direction: alternate;
}
```

...could be replaced by using the `animation` shorthand.

```css
p {
  animation: 3s infinite alternate slidein;
}
```

To learn more about the sequence in which different animation property values can be specified using the `animation` shorthand, see the {{cssxref("animation")}} reference page.

> 

Reviewer: should we add a warning to recommend putting the name last, since if a keyframe name is a keyword, the browser assumes the first instance is the keyword, not the name.

Author: Text to this effect is present on the [animation](https://developer.mozilla.org/en-US/docs/Web/CSS/animation#syntax) page, which is where the link is pointing.
(The sequence of sections on the 'animation' page does not seem consistent with other pages. I'll open another PR to fix that.)

## Configuring the animation

To create a CSS animation sequence, you style the element you want to animate with the {{cssxref("animation")}} property or its sub-properties. This lets you configure the timing, duration, and other details of how the animation sequence should progress. This does **not** configure the actual appearance of the animation, which is done using the {{cssxref("@keyframes")}} at-rule as described in the [Defining the animation sequence using keyframes](#defining_the_animation_sequence_using_keyframes) section below.

The sub-properties of the {{cssxref("animation")}} property are:

> - {{cssxref("animation-composition")}}

Reviewer: I don't think this is part of the `animation` shorthand yet. it's still [<time>](https://drafts.csswg.org/css-values-3/#time-value) [||](https://drafts.csswg.org/css-values-4/#comb-any) [<easing-function>](https://drafts.csswg.org/css-easing/#typedef-easing-function) || <time> || [<single-animation-iteration-count>](https://drafts.csswg.org/css-animations/#typedef-single-animation-iteration-count) || [<single-animation-direction>](https://drafts.csswg.org/css-animations/#typedef-single-animation-direction) || [<single-animation-fill-mode>](https://drafts.csswg.org/css-animations/#typedef-single-animation-fill-mode) || [<single-animation-play-state>](https://drafts.csswg.org/css-animations/#typedef-single-animation-play-state) || [ none [|](https://drafts.csswg.org/css-values-4/#comb-one) [<keyframes-name>](https://drafts.csswg.org/css-animations/#typedef-keyframes-name) ] per the spec.

  - : Specifies the {{Glossary("Composite operation")}} to use when multiple animations affect the same property simultaneously.
- {{cssxref("animation-delay")}}

>   - : Specifies the delay between an element loading and the start of an animation sequence.

Reviewer: ```suggestion
  - : Specifies the delay between an element loading and the start of an animation sequence, or if the animation should start immediately, but mid-animation.
```

Author: ", but mid-animation" bit is not clear

- {{cssxref("animation-direction")}}

>   - : Specifies whether the animation should alternate direction on each run through the sequence or reset to the start point and repeat.

Reviewer: ```suggestion
  - : Specifies whether the animation's first iteration should be forward or backward and whether subsequent iterations should alternate direction on each run through the sequence or reset to the start point and repeat.
```

- {{cssxref("animation-duration")}}
  - : Specifies the length of time in which an animation completes one cycle.
- {{cssxref("animation-fill-mode")}}

>   - : Specifies the values that are applied by the animation before and after it runs.

Reviewer: ```suggestion
  - : Specifies whether keyframe values are applied by the animation before and after it runs.
```

- {{cssxref("animation-iteration-count")}}
  - : Specifies the number of times an animation should repeat.
- {{cssxref("animation-name")}}
  - : Specifies the name of the {{cssxref("@keyframes")}} at-rule describing the animation's keyframes.
- {{cssxref("animation-play-state")}}

>   - : Specifies whether to pause and resume an animation sequence.

Reviewer: ```suggestion
  - : Specifies whether to pause or play an animation sequence.
```

- {{cssxref("animation-timing-function")}}
  - : Specifies how an animation transitions through keyframes by establishing acceleration curves.
<!--
- {{cssxref("animation-timeline")}}
  - : Specifies the names of one or more {{cssxref("@scroll-timeline")}} at-rules describing the scroll animations.
-->

## Defining the animation sequence using keyframes

> After you've configured the animation's timing, you need to define the appearance of the animation. This is done by establishing two or more keyframes using the {{cssxref("@keyframes")}} at-rule. Each keyframe describes how the animated element should render at a given time during the animation sequence.

Reviewer: ```suggestion
After you've configured the animation's timing, you need to define the appearance of the animation. This is done by establishing one or more keyframes using the {{cssxref("@keyframes")}} at-rule. Each keyframe describes how the animated element should render at a given time during the animation sequence.
```

Reviewer: only one is needed, since 0% and 100% have the non-animated state as their values.

The CSS animation longhand properties can accept multiple values, separated by commas. This feature can be used when you want to apply multiple animations in a single rule and set different durations, iteration counts, etc., for each of the animations. Let's look at some quick examples to explain the different permutations.

In this first example, three animation names are set, but there's only one duration and iteration count. In this case, all three animations are given the same duration and iteration count.

```css
animation-name: fadeInOut, moveLeft300px, bounce;
animation-duration: 3s;
animation-iteration-count: 1;
```

In this second example, there are three duration and three iteration count values. In this case, each animation is assigned a value of duration and iteration count with the same position as the animation name. So the `fadeInOut` animation is assigned a duration of `2.5s` and an iteration count of `2`, and the `bounce` animation is assigned a duration of `1s` and an iteration count of `5`.

```css
animation-name: fadeInOut, moveLeft300px, bounce;
animation-duration: 2.5s, 5s, 1s;
animation-iteration-count: 2, 1, 5;
```

> In this third example, three animations are specified, but only two durations and iteration counts. In such cases where there are not enough values in the list to assign a separate one to each animation, the value assignment cycles from the first to the last item in the available list and then resets to the first tem. So, `fadeInOut` gets a duration of `2.5s`, and `moveLeft300px` gets a duration of `5s`, which is the last value in the list of duration values. The duration value assignment now resets to the first value; `bounce`, therefore, gets a duration of `2.5s`. The iteration count values (and any other property values you specify) will be assigned in the same way.

Reviewer: ```suggestion
In this third example, three animations are specified, but only two durations and iteration counts. In such cases where there are not enough values in the list of durations to assign a separate one to each animation, the value assignment cycles from the first to the last item in the available list and then resets to the first item. So, `fadeInOut` gets a duration of `2.5s`, and `moveLeft300px` gets a duration of `5s`, which is the last value in the list of duration values. The duration value assignment now resets to the first value; `bounce`, therefore, gets a duration of `2.5s`. The iteration count values (and any other property values you specify) will be assigned in the same way.
```

Reviewer: Let's add a title so we can link directly to this title from the five animation pages instead of:

> see [Setting multiple animation property values](/en-US/docs/Web/CSS/CSS_Animations/Using_CSS_animations#setting_multiple_animation_property_values).

So they can go directly here instead of reading thru the three examples,

Reviewer: It would be good to give this section and #### title, so we can link to it directly from the five animation pages where we have:

> see [Setting multiple animation property values](/en-US/docs/Web/CSS/CSS_Animations/Using_CSS_animations#setting_multiple_animation_property_values).

Author: This is a great suggestion. As I started working through it, I ended up pulling out "Using the animation shorthand" and "Setting multiple animation property values" sections from inside "Examples" and turned them from ### to ##.

I started to add ###s under "Setting multiple animation property values" as you've suggested but it seems two cases in the "Setting multiple animation property values" section are applicable in the other animation pages where we have that note. So it might not be worth adding the ###s and linking to just the last case.

BTW, thank you for suggesting the text for the inverse case 👍.
